pkg/asset/metadata/metadata.go

```diff
@@ -0,0 +1,80 @@
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+
+	// "github.com/hashicorp/terraform/terraform"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+)
+
+const (
+	// MetadataFilename is name of the file where clustermetadata is stored.
+	MetadataFilename  = "metadata.json"
+	metadataAssetName = "Cluster Metadata"
+)
+
+// Metadata depends on cluster and installconfig,
+type Metadata struct {
+	installConfig asset.Asset
+	cluster       asset.Asset
+}
+
+var _ asset.Asset = (*Metadata)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (m *Metadata) Name() string {
+	return metadataAssetName
+}
+
+// Dependencies returns the dependency of the MetaData.
+func (m *Metadata) Dependencies() []asset.Asset {
+	return []asset.Asset{m.installConfig, m.cluster}
+}
+
+// Generate generates the metadata.yaml file.
+func (m *Metadata) Generate(parents map[asset.Asset]*asset.State) (*asset.State, error) {
+	installCfg, err := installconfig.GetInstallConfig(m.installConfig, parents)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get install config state in the parent asset states")
+	}
+
+	// TODO: might have to read the tfstate
+	// buf := bytes.NewBuffer(parents[m.cluster].Contents[0].Data)
+	// tfState, err := terraform.ReadState(buf)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	cm := &types.ClusterMetadata{
+		ClusterName: installCfg.Name,
+	}
+	if installCfg.Platform.AWS != nil {
+		cm.ClusterPlatformMetadata.AWS = &types.ClusterAWSPlatformMetadata{
+			Region: installCfg.Platform.AWS.Region,
+			Identifier: map[string]string{
+				"tectonicClusterID": installCfg.ClusterID,
+			},
+		}
+	} else if installCfg.Platform.Libvirt != nil {
+		cm.ClusterPlatformMetadata.Libvirt = &types.ClusterLibvirtPlatformMetadata{
+			URI: installCfg.Platform.Libvirt.URI,
+		}
+	}
```

Comment:
`else` error out?  Or are we comfortable guarding against "forgot to set a platform" elsewhere?

Reply:
changed to `switch`